components/discord_bot/actions/modify-guild-member/modify-guild-member.mjs

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-unused-vars */
+import constants from "../../constants.mjs";
+import common from "../common.mjs";
+
+const { discord } = common.props;
+
+const {
+  GUILD_TEXT,
+  GUILD_NEWS,
+  ...OTHER_CHANNELS
+} = constants.CHANNEL_TYPES;
+
+const NOT_ALLOWED_CHANNELS =
+  Object.keys(OTHER_CHANNELS).map((key) => OTHER_CHANNELS[key]);
+
+export default {
+  ...common,
+  key: "discord_bot-modify-guild-member",
+  name: "Modify Guild Member",
+  description: "Update attributes of a guild member. [See the docs here](https://discord.com/developers/docs/resources/guild#modify-guild-member)",
+  type: "action",
+  version: "0.0.1",
+  props: {
+    ...common.props,
+    userId: {
+      propDefinition: [
+        discord,
+        "userId",
+        ({ guildId }) => ({
+          guildId,
+        }),
+      ],
+    },
+    nick: {
+      propDefinition: [
+        discord,
+        "nick",
+      ],
+      optional: true,
+    },
+    roles: {
+      propDefinition: [
+        discord,
+        "roleId",
+        ({
+          guildId, userId,
+        }) => ({
+          guildId,
+          userId,
+          isAddRole: true,
+        }),
+      ],
+      type: "string[]",
+      optional: true,
+    },
+    mute: {
+      type: "boolean",
+      label: "Mute",
+      description: "Whether the user is muted in voice channels. Will throw a 400 error if the user is not in a voice channel",
+      optional: true,
+    },
+    deaf: {
+      type: "boolean",
+      label: "Deaf",
+      description: "Whether the user is deafened in voice channels. Will throw a 400 error if the user is not in a voice channel",
+      optional: true,
+    },
+    channelId: {
+      description: "Id of channel to move user to (if they are connected to voice)",
+      propDefinition: [
+        discord,
+        "channelId",
+        ({ guildId }) => ({
+          guildId,
+          notAllowedChannels: NOT_ALLOWED_CHANNELS,
+        }),
+      ],
+      optional: true,
+    },
+    comunicationDisabledUntil: {
+      type: "string",
+      label: "Comunication Disabled Until",
+      description: "When the user's [timeout](https://support.discord.com/hc/en-us/articles/4413305239191-Time-Out-FAQ) will expire and the user will be able to communicate in the guild again (up to 28 days in the future), set to null to remove timeout. Will throw a 403 error if the user has the ADMINISTRATOR permission or is the owner of the guild",
```

Comment:
Can you add examples of how the user could fill that field? e.g: `1 hour` or `10 days` or the way its supposed to be filled?